src/use-queries-state.ts

```diff
@@ -0,0 +1,31 @@
+import type { MaybeRefOrGetter } from 'vue'
+import { computed, toValue } from 'vue'
+import type { EntryKey } from './entry-options'
+import type { QueryStatus } from './query-store'
+import { useQueryCache } from './query-store'
+
+interface QueryState {
+    data: unknown
+    error: unknown
+    status: QueryStatus
+}
+
+// TODO: accept also a since key (instead of an array) as argument
+// TODO: accept a ref for `keys` (to handle adding/removing entries for example)?
+export function useQueriesState(keys: Array<MaybeRefOrGetter<EntryKey>>) {
+  const { ensureEntry } = useQueryCache()
+  return computed(() => {
+  return keys.map<QueryState[]>(
+    // TODO: fix TS issue
+    (key) => {
+        const query = ensureEntry(toValue(key))
```

Comment:
Should we add and use the `findQuery` function here? It will work if we decide not to create a query in `useQueriesState` in case it has not been created